### ISSUE TITLE
Normalised embeddings when using a combination of miners and losses.

### ISSUE THREAD

**Issue by riririririi** [KevinMusgrave/pytorch-metric-learning#682]
hi, first of all thank you for making this repo and maintaining it, you have been a great help.
i am training a feature extractor to cluster similar objects based on visual similarity, and i am using a combination of triplet loss with euclidean distance and cosine similarity, here is the relevant code below:   
```
batch_miner_cosine=miners.BatchEasyHardMiner(pos_strategy="all", neg_strategy="all", distance=distances.CosineSimilarity())
batch_miner_euclidean=miners.BatchEasyHardMiner(pos_strategy="all", neg_strategy="all", distance=distances.LpDistance())

triplet_loss_cosine=losses.TripletMarginLoss(margin=0.1, distance=distances.CosineSimilarity())
triplet_loss_euclidean=losses.TripletMarginLoss(margin=0.1, distance=distances.LpDistance())

writer = SummaryWriter(log_dir="/pytorch_metric_clustering_trial/weights_multiloss/resnet-mod-3x3_weights/logs")

for epoch in range(start_epoch, 100001):
    running_loss = 0.0
    current_lr = optimizer.param_groups[0]['lr']

    for i, data in enumerate(train_loader):
        inputs, labels = data
        inputs = inputs.to(device)
        labels = labels.to(device)
        optimizer.zero_grad()
        outputs = model(inputs)

        mined_outputs_cosine = batch_miner_cosine(outputs, labels)
        mined_outputs_euclidean = batch_miner_euclidean(outputs, labels)

        loss_euclidean = triplet_loss_euclidean(outputs, labels, mined_outputs_euclidean)
        loss_cosine = triplet_loss_cosine(outputs, labels, mined_outputs_cosine)

        loss = (0.9*loss_cosine) + (0.1*loss_euclidean)

        writer.add_scalar("Loss/train", loss, epoch)
        loss.backward()
        optimizer.step()
        running_loss += loss.item()
        if i % 2 == 0:
            print(f"At epoch = {epoch}, iter = {i} out of total {len(train_loader)}, loss = {loss:.16f}, current_lr = {current_lr}", end='\r')
    
    epoch_loss = running_loss / len(train_loader)
    loss_history.append(epoch_loss)
    writer.flush()
```
in the documentation for the BatchEasyHardMiner and TripletMarginLoss, the default distance is given like "[LpDistance(normalize_embeddings=True, p=2, power=1)]"
i also have added an L2 normalising layer in my model after the final layer, so the values that the miner gets are already normalised, and i am confused on how to turn off the normalisation if it is on by default, so that i could use my model as it is.

**Comment by KevinMusgrave** [comment]
Here's how to turn off normalization:

```python
from pytorch_metric_learnings import losses, miners, distances

loss_dist = distances.LpDistance(normalize_embeddings=False)
loss_fn = losses.TripletMarginLoss(distance=loss_dist)

miner_dist = distances.LpDistance(normalize_embeddings=False)
miner_fn = miners.BatchEasyHardMiner(distance=miner_dist)
```

**Comment by riririririi** [comment]
Thanks a lot for your help !
